libs/server/Storage/Session/ObjectStore/SetOps.cs

```
@@ -422,6 +422,144 @@ internal unsafe GarnetStatus SetMove(ArgSlice sourceKey, ArgSlice destinationKey
+            {
+                var members = SetIntersect(keys, ref setObjectStoreLockableContext);
+
+                var newSetObject = new SetObject();
+                foreach (var item in members)
+                {
+                    _ = newSetObject.Set.Add(item);
+                    newSetObject.UpdateSize(item);
+                }
+                _ = SET(key, newSetObject, ref setObjectStoreLockableContext);
+                count = members.Count;
+            }
+            finally
+            {
+                if (createTransaction)
+                    txnManager.Commit(true);
+            }
+
+            return GarnetStatus.OK;
+        }
+
+
+        private HashSet<byte[]> SetIntersect<TObjectContext>(ArgSlice[] keys, ref TObjectContext objectContext)
+           where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, SpanByte, GarnetObjectStoreOutput, long>
+        {
+            var result = new HashSet<byte[]>(new ByteArrayComparer());
+            if (keys.Length == 0)
+            {
+                return result;
+            }
+
+            var status = GET(keys[0].ToArray(), out var first, ref objectContext);
+            if (status == GarnetStatus.OK)
+            {
+                if (first.garnetObject is SetObject firstObject)
+                {
+                    result = new HashSet<byte[]>(firstObject.Set, new ByteArrayComparer());
+                }
+            }
+            else
+            {
+                return result;
+            }
+
+
+            for (var i = 1; i < keys.Length; i++)
+            {
+                // intersection of anything with empty set is empty set
+                if (result.Count == 0)
+                {
+                    return result;
+                }
+
+                status = GET(keys[i].ToArray(), out var next, ref objectContext);
+                if (status == GarnetStatus.OK)
+                {
+                    if (next.garnetObject is SetObject nextObject)
```

If you encounter an object that is not a SetObject you should return an error (CmdStrings.RESP_ERR_WRONG_TYPE)

---

As far as I can see, at this point there is no easy way to implement this. There is a separate but not yet merged branch in which there is initial support for RESP_ERR_WRONG_TYPE
branch name :obj-store-wrongtype